src/KK.DotNet.BackgroundTasks.Scheduled/Scheduler.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace KK.DotNet.BackgroundTasks.Scheduled
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using Cronos;
+    using Microsoft.Extensions.Logging;
+
+    public class Scheduler
+    {
+        private readonly ILogger<Scheduler> logger;
+        private readonly List<SchedulerTask> schedulerTasks = new List<SchedulerTask>();
+        public ReadOnlyCollection<SchedulerTask> Tasks => this.schedulerTasks.AsReadOnly();
+
+        public Scheduler(
+           ILogger<Scheduler> logger
+        ) => this.logger = logger;
+
+        public event EventHandler TaskAdded;
+        protected virtual void OnTaskAdded() => TaskAdded?.Invoke(this, EventArgs.Empty);
+
+        public void AddTask(IScheduledTask scheduledTask)
+        {
+            var taskName = string.IsNullOrWhiteSpace(scheduledTask.Options.Name) ? Guid.NewGuid().ToString() : scheduledTask.Options.Name;
```

Comment:
Many other lines have been broken before (which I think is optional), but in this case I guess it makes a lot of sense:

```cs
var taskName = string.IsNullOrWhiteSpace(scheduledTask.Options.Name) 
                  ? Guid.NewGuid().ToString() 
                  : scheduledTask.Options.Name;
```